Kill plugin script

danwinship opened the pull request:
The CNI port already got us most of the way there. This moves the remaining bits of pod setup/teardown into go rather than shell script.

(I started this a long time ago, and finished it now because of #11990.)

@openshift/networking PTAL

smarterclayton commented on pkg/sdn/plugin/pod_linux.go:
```diff
@@ -254,20 +190,97 @@ func (m *podManager) ipamDel(id string) error {
+	return otx.EndTransaction()
+}
+
+func (m *podManager) setupPodBandwidth(pod *kapi.Pod, hostVeth string) error {
```
Are these methods actually related?  If all you're accessing is `m.ovs` taking ovs as an argument is just as good.  We generally lean from a style perspective on not creating grouping types unless there is an actual "group".

danwinship commented:
@smarterclayton like that?

smarterclayton commented on pkg/sdn/plugin/pod_linux.go:
```diff
@@ -254,20 +191,88 @@ func (m *podManager) ipamDel(id string) error {
+	return otx.EndTransaction()
+}
+
+func setupPodBandwidth(ovsif *ovs.Interface, pod *kapi.Pod, hostVeth string) error {
```
Yeah, looks good.  I find a lot of the time it's easy to "over OO" stuff, and I think for many functions you end up with more testable and less coupled code.

dcbw commented:
@danwinship So generally LGTM, though we'll have the same upgrade problem when a master+node host is updated, since the RPM changes remove the script.  Then pods on that node will fail because the script doesn't exist.  Maybe we just keep the script around in the RPM (I'm pretty sure we don't  need it in the images, just in the specfile) for one more release?

danwinship commented:
> Maybe we just keep the script around in the RPM (I'm pretty sure we don't need it in the images, just in the specfile) for one more release?

openshift doesn't care where the script is as long as it's in $PATH, so we could have the update process copy it from /usr/bin to /usr/local/bin before upgrading the RPM, and then delete it after restarting the node service

danwinship commented:
(re-push is just to squash the fixup)

smarterclayton commented:
Generally if it's more complex to do it in ansible, doing it in your code
(preserving the script for at least one version) reduces the impact of this
change to other teams (keeps your dirty laundry secret)

On Wed, Dec 7, 2016 at 10:41 AM, Dan Winship <notifications@github.com>
wrote:

> (re-push is just to squash the fixup)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/12145#issuecomment-265480999>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_p0HAVONxBM1G6FQSCNoLgZcwZ-ULks5rFtOhgaJpZM4LEqyc>
> .
>

knobunc commented:
Do we only support single version upgrades?  I.e. to get from 3.2 to 3.4
you need to go through 3.3 (and start it up)?

On Wed, Dec 7, 2016 at 11:46 AM, Clayton Coleman <notifications@github.com>
wrote:

> Generally if it's more complex to do it in ansible, doing it in your code
> (preserving the script for at least one version) reduces the impact of this
> change to other teams (keeps your dirty laundry secret)
>
> On Wed, Dec 7, 2016 at 10:41 AM, Dan Winship <notifications@github.com>
> wrote:
>
> > (re-push is just to squash the fixup)
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/openshift/origin/pull/12145#issuecomment-265480999>,
> > or mute the thread
> > <https://github.com/notifications/unsubscribe-auth/ABG_
> p0HAVONxBM1G6FQSCNoLgZcwZ-ULks5rFtOhgaJpZM4LEqyc>
> > .
>
> >
>
> —
> You are receiving this because you are on a team that was mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/12145#issuecomment-265500850>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AJkavRFjQG2TZAtNHdAHnQ4LkNZbTmvhks5rFuLVgaJpZM4LEqyc>
> .
>

danwinship commented:
> (keeps your dirty laundry secret)

It's not clear whose dirty laundry it is; leaving openshift running while changing the rpm out from under it is kind of dirty on their part. But sure, we can leave the script around.

> Do we only support single version upgrades?

yes

smarterclayton commented:
Not quite true.  We support single version no downtime upgrades.  We do not
support multi version no downtime (of the control plane) upgrades, although
anything beyond the control plane needs an exception.

On Dec 7, 2016, at 12:22 PM, Dan Winship <notifications@github.com> wrote:

(keeps your dirty laundry secret)

It's not clear whose dirty laundry it is; leaving openshift running while
changing the rpm out from under it is kind of dirty on their part. But
sure, we can leave the script around.

Do we only support single version upgrades?

yes

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/openshift/origin/pull/12145#issuecomment-265511764>,
or mute
the thread
<https://github.com/notifications/unsubscribe-auth/ABG_p6ocOKSZ5Vfgq3AcGEiGoxq4dQXeks5rFutHgaJpZM4LEqyc>
.

danwinship commented:
[test][testextended][extended:networking]

danwinship commented:
Are we going to try to get this in for 3.5? (It's not a feature so I guess it's not subject to the "feature freeze" deadline anyway, but...)

danwinship commented:
Rebased and repushed now that the tree is open for 3.6. Also changed it to make it continue installing the now-unused script, to avoid problems during upgrade. We can remove it completely for 3.7.

danwinship commented:
flake #12855, [testextended]

danwinship commented:
@openshift/networking the test failures will be fixed by #13061 when it lands. This is ready for review.

pravisankar commented on pkg/sdn/plugin/pod_linux.go:
```diff
@@ -252,20 +191,83 @@ func (m *podManager) ipamDel(id string) error {
+	}
+
+	if ovsEgress > 0 {
+		qos, err := ovsif.Create("qos", "type=linux-htb", fmt.Sprintf("other-config:max-rate=%d", ovsEgress))
```
Where is the code equivalent to "ip link set dev ${veth_host} qlen 1000" in openshift-sdn-ovs?

danwinship replied:
ah, i think this branch originally predates that addition and I must have missed it when rebasing

pravisankar commented on pkg/sdn/plugin/pod_linux.go:
```diff
@@ -252,20 +191,83 @@ func (m *podManager) ipamDel(id string) error {
+		ovsEgress = podIngress.Value()
+	}
+	if podEgress != nil {
+		ovsIngress = podEgress.Value() / 1024
```
is it /1024 or /1000? 1024 might be right (we are using 1000 in openshift-sdn-ovs)

danwinship replied:
Oh, yeah, I saw 1000 in openshift-sdn-ovs but assumed it was probably wrong and should be 1024. @dcbw?

danwinship replied:
ovs docs say "ingress_policing_rate: ... Maximum rate for data received on this interface, in kbps." so 1024 is correct

pravisankar commented:
Added couple of comments, rest of the code LGTM.
Need to resolve some merge conflicts

danwinship commented:
Rebased and added in the qlen thing (and did another check to see if I'd missed anything when rebasing, and didn't see anything). It seems that the netlink library used for all the other /sbin/ip stuff in pod_linux.go doesn't let you do "set qlen" (even if we updated to the latest git) so I had to use ipcmd instead.

danwinship commented:
[merge]
